external/loaders/loaders/mappers/_hybrid.py

```diff
@@ -313,52 +321,120 @@ def _add_fine_res_inputs(fine: xarray.Dataset, input_feature_url: Optional[str]
+
+    return _add_fine_res_inputs(fine, input_feature_url)
+
+
+def _extend_lower(fine: xarray.Dataset, vertical_dim: str = "z") -> xarray.Dataset:
+    fine_sources = get_variables(fine, ["Q1", "Q2"])
+    fine_sources_new_bottom = fine_sources.isel({vertical_dim: -2})
+    fine_sources_without_bottom = fine_sources.isel({vertical_dim: slice(None, -2)})
+    fine_sources_extended_lower = xarray.concat(
+        [fine_sources_without_bottom, fine_sources_new_bottom], dim=vertical_dim
     )
+    return fine_sources_extended_lower.Q1, fine_sources_extended_lower.Q2
 
 
-@mapper_functions.register
-def open_precomputed_fine_resolution(
-    fine_url: str, input_feature_url: Optional[str] = None
-) -> GeoMapper:
-    """
-    Open the fine-res mapper from precomputed data, optionally using state
-        from another run.
+def open_fine_resolution_extend_lower(
+    fine_url: str, input_feature_url: str, include_temperature_nudging: bool
+) -> xarray.Dataset:
+
+    fine = _compute_fine_res_portion(fine_url, include_temperature_nudging)
+    fine["Q1"], fine["Q2"] = _extend_lower(fine)
 
+    return _add_fine_res_inputs(fine, input_feature_url)
+
+
+FINE_RESOLUTION_APPROACHES = [
+    "apparent_sources_only",
+    "precomputed_apparent_sources",
+    "apparent_sources_plus_nudging",
+    "apparent_sources_plus_dynamics_differences",
+    "apparent_sources_extend_lower",
+]
+
+
+@dataclass
+class FineResolutionOpenerConfig:
+    approach: str
+    fine_url: str
+    input_feature_url: Optional[str] = None
+    additional_source_url: Optional[str] = None
+    include_temperature_nudging: bool = False
+
+
+@mapper_functions.register
+def open_fine_resolution(approach: Optional[str] = None, **kwargs: dict):
+    f"""
+    Open the fine-res mapper using several configuration options
+    
     Args:
-        fine_url: url where coarsened fine resolution data is stored, must include
-            precomputed Q1 and Q2
+        approach: one of a set of available approaches, including
+            {FINE_RESOLUTION_APPROACHES}; if not specified will use
+            'apparent_sources_only'; usage of the kwargs below will depend on
+            `approach`
+        fine_url: url where coarsened fine resolution data is stored
         input_feature_url: url to fv3gfs_run to use for state inputs. If not provided,
             the state of the fine-res data will be used as input.
-
+        additional_source_url: url of additional temperature and moisture tendencies
+            (e.g., nudging or dynamics differences) to be added to the fine resolution
+            data
+        include_temperature_nudging: whether to include fine-res nudging in Q1
+        
     Returns:
         a mapper
     """
-    return XarrayMapper(
-        open_precomputed_fine_resolution_dataset(
-            fine_url=fine_url, input_feature_url=input_feature_url
-        )
+    if approach is None:
+        approach = "apparent_sources_only"
+
+    fine_res_config = dacite.from_dict(
```

Comment:
```suggestion
    config = dacite.from_dict(
```
the `fine_res` seems a bit redundant here.